bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -1677,19 +1689,19 @@ bool BCStateTran::onMessage(const AskForCheckpointSummariesMsg *m, uint32_t msgL
                                                        msg->requestMsgSeqNum,
                                                        msg->sizeofRvbData()));
 
-    replicaForStateTransfer_->sendStateTransferMessage(reinterpret_cast<char *>(msg), msg->size(), toReplicaId);
-    CheckpointSummaryMsg::free(msg);
+    replicaForStateTransfer_->sendStateTransferMessage(
+        reinterpret_cast<char *>(msg.getBytes()), msg->size(), toReplicaId);
```

Comment:
see my message below, not sur why we dont use the internal buffer as char* and call this function data():
`replicaForStateTransfer_->sendStateTransferMessage(msg.data(), msg->size(), toReplicaId);`